compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -175,28 +229,71 @@ struct GPUSetEncodingOpLoweringConversion
-    //    ACC: 256 -> 16x16 (innerTiles[0]xinnerTiles[1])
-
-    // TODO(lialan): Replace the op with the tensor.expand_shape op.
-    rewriter.replaceOp(encodingOp, packOp->getResult());
+    // Create expand_shape, to tile the inner most two dimensions.
+    auto sourceShape = packOp->getResult().getType().getShape();
+    assert(intrinsicVectorShape.size() == 2); // TODO: relax this
+    auto iT1 = intrinsicVectorShape[0];
+    auto iT2 = intrinsicVectorShape[1];
+    auto oT1 = sourceShape[2] / iT1;
+    auto oT2 = sourceShape[3] / iT2;
+    SmallVector<int64_t> expandShapeShape = {
+        sourceShape[0], sourceShape[1], oT1, iT1, oT2, iT2};
+    assert(expandShapeShape.size() == 6);
+    auto expandShapeType = RankedTensorType::get(
+        expandShapeShape, encodingOp.getSourceType().getElementType());
+
+    SmallVector<ReassociationIndices> reassociationMap = {
+        {0}, {1}, {2, 3}, {4, 5}};
+    auto expandShapeOp = rewriter.create<tensor::ExpandShapeOp>(
+        loc, expandShapeType, packOp->getResult(), reassociationMap);
+
+    // create linalg.transpose on expandShapeShape
+    size_t origRank = origRank = encodingOp.getSourceType().getRank();
+    SmallVector<int64_t> transposeResultDims;
+    transposeResultDims.push_back(expandShapeShape[0]);
+    transposeResultDims.push_back(expandShapeShape[1]);
+    for (int i = 0; i < transposePermutation.size(); i++) {
+      transposeResultDims.push_back(
+          expandShapeShape[origRank + transposePermutation[i]]);
+    }
+    SmallVector<int64_t> newTransposePermutation(origRank +
+                                                 transposePermutation.size());
+    newTransposePermutation[0] = 0;
+    newTransposePermutation[1] = 1;
+    for (int i = 0; i < transposePermutation.size(); i++) {
+      newTransposePermutation[origRank + i] =
+          origRank + transposePermutation[i];
+    }
+    auto emptyTensor = rewriter.create<tensor::EmptyOp>(
+        loc, transposeResultDims, encodingOp.getSourceType().getElementType());
+    auto transposeOp = rewriter.create<linalg::TransposeOp>(
+        loc, expandShapeOp, emptyTensor, newTransposePermutation);
+
+    // We want to make the shape consistent, so we need to append it with a
+    // `collapse_shape` and a `expand_shape`, just to be conformant with how we
+    // materialize for Flow and HAL ops:
+    // 1. collapse tiled dimensions into one dim
+    SmallVector<ReassociationIndices> collapseReassoc = {
+        {0}, {1}, {2, 3, 4, 5}};
+    SmallVector<int64_t> collapsedShape = {sourceShape[0], sourceShape[1],
+                                           sourceShape[2] * sourceShape[3]};
+    RankedTensorType revertShapeType = RankedTensorType::get(
+        collapsedShape, encodingOp.getSourceType().getElementType());
+
+    auto collapseShapeOp = rewriter.create<tensor::CollapseShapeOp>(
+        loc, revertShapeType, transposeOp->getResult(0),
+        ArrayRef<ReassociationIndices>(collapseReassoc));
```

Comment:
I think passing `collapseReassoc` directly should work?

Reply:
You are right! it can calculate the reassociation for you